code/game/objects/structures/headpike.dm

```diff
@@ -0,0 +1,39 @@
+/obj/structure/headpike
+	name = "spooky head on a spear"
+	desc = "When you really want to send a message."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "headpike"
+	density = 0
+	anchored = 1
+
+/obj/structure/headpike/bone //for bone spears
+	icon_state = "headpike-bone"
+
+
+/obj/structure/headpike/CheckParts(list/parts_list)
+	..()
+	var/obj/item/bodypart/head/H = locate() in contents
+	if(H)
+		overlays = H.overlays //head
```

Comment:
2 things.

1. this line should be in `update_icon()`
2. It won't work with overlay_queuing. You need `copy_overlays(H, TRUE)`